src/duration.rs

```diff
@@ -681,81 +892,73 @@ fn time_unit() {
     let third_hour: Duration = -1 * TimeUnit::Hour / 3;
     let sum: Duration = quarter_hour + third_hour;
     let delta = sum.in_unit(TimeUnit::Millisecond).floor()
-        - sum.in_unit(TimeUnit::Second).floor() * TwoFloat::from(1000.0);
-    println!("{:?}", delta * TwoFloat::from(-1) == TwoFloat::from(0));
+        - sum.in_unit(TimeUnit::Second).floor() * 1000.0;
+    println!("{:?}", delta * -1.0 == 0.0);
     assert!((sum.in_unit_f64(TimeUnit::Minute) + 35.0).abs() < EPSILON);
 }
 
 #[test]
 fn duration_print() {
     // Check printing adds precision
     assert_eq!(
-        format!("{}", TimeUnit::Day * 10.0 + TimeUnit::Hour * 5),
-        "10 days 5 h 0 min 0 s"
+        format!("{}", TimeUnit::Day * 10.0 + TimeUnit::Hour * 5).trim(),
+        "10 days 5 h"
     );
 
     assert_eq!(
-        format!("{}", TimeUnit::Hour * 5 + TimeUnit::Millisecond * 256),
+        format!("{}", TimeUnit::Hour * 5 + TimeUnit::Millisecond * 256).trim(),
         "5 h 0 min 0 s 256 ms"
     );
 
     assert_eq!(
         format!(
             "{}",
             TimeUnit::Hour * 5 + TimeUnit::Millisecond * 256 + TimeUnit::Nanosecond
-        ),
-        "5 h 0 min 0 s 256 ms 1 ns"
+        ).trim(),
+        "5 h 0 min 0 s 256 ms 0 us 1 ns"
     );
 
     assert_eq!(
-        format!("{}", TimeUnit::Hour + TimeUnit::Second),
+        format!("{}", TimeUnit::Hour + TimeUnit::Second).trim(),
         "1 h 0 min 1 s"
     );
 
-    assert_eq!(
-        format!(
-            "{}",
-            TimeUnit::Hour * 5
-                + TimeUnit::Millisecond * 256
-                + TimeUnit::Microsecond
-                + TimeUnit::Nanosecond * 3.5
-        ),
-        "5 h 0 min 0 s 256 ms 1003.5 ns"
-    );
-
     // Check printing negative durations only shows one negative sign
     assert_eq!(
-        format!("{}", TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256),
+        format!("{}", TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256).trim(),
         "-5 h 0 min 0 s 256 ms"
     );
 
+    let d : Duration = TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256 + TimeUnit::Nanosecond * -3;
+    dbg!(d.in_seconds());
+
     assert_eq!(
         format!(
             "{}",
-            TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256 + TimeUnit::Nanosecond * -3.5
-        ),
-        "-5 h 0 min 0 s 256 ms 3.5 ns"
+            TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256 + dbg!(TimeUnit::Nanosecond * -3)
+        ).trim(),
+        "-5 h 0 min 0 s 256 ms 0 us 3 ns"
     );
 
     assert_eq!(
         format!(
             "{}",
             (TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256)
                 - (TimeUnit::Hour * -5 + TimeUnit::Millisecond * -256 + TimeUnit::Nanosecond * 2)
-        ),
+        ).trim(),
         "-2 ns"
     );
 
     assert_eq!(
-        format!("{}", Duration::from_nanoseconds(TwoFloat::from(2))),
+        format!("{}", Duration::from_nanoseconds_f(2.0)).trim(),
```

Comment:
I wonder if the `from_nanoseconds` shouldn't be generic over some num_trait type to avoid the `_f` and `_u`. Thoughts?

Reply:
Sounds good, we can make this a new issue.